File: src/main/java/no/nav/pensjon/innsyn/domain/Opptjening.java
package no.nav.pensjon.innsyn.domain;

public class Opptjening {

    private final String type;
    private final String status;
    private final int opptjeningsar;
    private final double pensjonsgivendeInntekt;
    private final double poeng;
    private final double uforegrad;

    public Opptjening(String type,
                      String status,
                      int opptjeningsar,
                      double pensjonsgivendeInntekt,
                      double poeng,
                      double uforegrad) {
        this.type = type;
        this.status = status;
        this.opptjeningsar = opptjeningsar;
        this.pensjonsgivendeInntekt = pensjonsgivendeInntekt;
        this.poeng = poeng;
        this.uforegrad = uforegrad;
    }

    public String getType() {
        return type;
    }

    public String getStatus() {
        return status;
    }

    public int getOpptjeningsar() {
        return opptjeningsar;
    }

    public double getPensjonsgivendeInntekt() {
        return pensjonsgivendeInntekt;
    }

    public double getPoeng() {
        return poeng;
    }

    public double getUforegrad() {
        return uforegrad;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/Opptjening.kt
package no.nav.pensjon.innsyn.domain

import javax.persistence.*

@Entity
@Table(name = "T_OPPTJN")
@SecondaryTables(
        SecondaryTable(name = "T_K_OPPTJN_T", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_OPPTJN_T")]),
        SecondaryTable(name = "T_K_OPPTJN_STATUS", pkJoinColumns = [PrimaryKeyJoinColumn(name = "K_OPPTJN_STATUS")])
)
data class Opptjening(
        @Column(name = "DEKODE", table = "T_K_OPPTJN_T")
        val type: String,
        @Column(name = "DEKODE", table = "T_K_OPPTJN_STATUS")
        val status: String,
        @Column(name = "OPPTJN_AR")
        val opptjeningsar: Int,
        @Column(name = "PGI_ANVENDT")
        val pensjonsgivendeInntekt: Double,
        @Column(name = "POENG")
        val poeng: Double,
        @Column(name = "MAX_UFOREGRAD")
        val uforegrad: Double
) {
    @Id
    @Column(name = "PERSON_ID_OPPTJN")
    var personId: Int? = null
}